src/fault-handler.jl

```diff
@@ -6,9 +6,27 @@ check_exited_exception(res::RemoteException) =
 check_exited_exception(res::ProcessExitedException) = true
 check_exited_exception(res) = false
 
+"""
+    handle_fault(...)
+
+An internal function to handle a worker dieing or being killed by the OS.
```

Comment:
you might want to use `dying` here as more common rather than `dieing` (very specialized term)

Reply:
Good catch :slightly_smiling_face:

Reply:
My first review is about dead stuff 💀 sob 🗡